### ISSUE TITLE
dubious ownership error in GitLab CI

### ISSUE THREAD

**Issue by gergelypolonkai** [akaihola/darker#524]
- [x] review akaihola/darkgraylib#74
- [x] merge akaihola/darkgraylib#74
- [ ] release `darkgraylib==2.0.1`
- [ ] update to `darkgraylib==2.0.1` in Darker

**Describe the bug**

While running darker in our GitLab CI, it fails with the following message:

```
ERROR:darker.git:fatal: detected dubious ownership in repository at '/builds/group/repo'
ERROR:darker.git:To add an exception for this directory, call:
ERROR:darker.git:
ERROR:darker.git:	git config --global --add safe.directory /builds/group/repo
```

**Environment (please complete the following information):**
 - OS: Kubernetes, running inside the official Python container
 - Python version 3.10.13
 - Git version 2.39.2
 - Darker version 1.7.2
 - Black version 22.8.0

**Additional context**

What makes this strange is that `git` itself works without a problem after marking the directory as safe. Here’s the output of some debug commands i put into our darker job:

```
$ git --version
git version 2.39.2
$ cat $HOME/.gitconfig || echo "No config"
cat: /project/.gitconfig: No such file or directory
No config
$ git status || echo "bah"
fatal: detected dubious ownership in repository at '/builds/group/repo'
To add an exception for this directory, call:
	git config --global --add safe.directory /builds/group/repo
bah
$ git config --global --add safe.directory /builds/group/repo
$ cat $HOME/.gitconfig || echo "No config"
[safe]
	directory = /builds/group/repo
$ git status || echo "bah"
HEAD detached at af72d68b1
nothing to commit, working tree clean
$ poetry run darker --diff --revision origin/master... app data_tests tests
ERROR:darker.git:fatal: detected dubious ownership in repository at '/builds/group/repo'
ERROR:darker.git:To add an exception for this directory, call:
ERROR:darker.git:
ERROR:darker.git:	git config --global --add safe.directory /builds/group/repo
```
The directory itself (`/builds/group/repo`) is owned by root and has permissions `0777`. The whole CI process is running as user `app` (UID 1000).

**Comment by akaihola** [comment]
Hi @gergelypolonkai,

Could you add `-vv` to Darker arguments to get more debug output?

**Comment by DeinAlptraum** [comment]
@akaihola Hello, I got the same error the other day, as described here: #590 
It seems that the `env=make_git_env()` argument passed to `subprocess.check_output` is too minimal of an environment. Simply removing this argument was enough for me to fix the issue, with darker running correctly and fixing files now.

**Comment by akaihola** [comment]
Hi @gergelypolonkai,

I'm trying to fix this issue in akaihola/darkgraylib#74. To test whether it helps, first install Darker and then on top of it:
```bash
pip install darkgraylib@git+https://github.com/akaihola/darkgraylib@git-dubious-ownership
```

I also invited you as a collaborator on Darkgraylib. Feel free to review akaihola/darkgraylib#74!
